ui/app/adapters/node-pool.js

```diff
@@ -14,4 +14,21 @@ export default class NodePoolAdapter extends ApplicationAdapter {
     resource = pluralize(resource);
     return `/v1/${relationshipResource}/${resource}`;
   }
+
+  findAll() {
+    return super.findAll(...arguments).catch((error) => {
+      // Handle the case where the node pool request is sent to a region that
+      // doesn't have node pools and the request is handled by the nodes
+      // endpoint.
+      const isNodeRequest = error.message.includes(
+        'node lookup failed: index error: UUID must be 36 characters'
```

Comment:
With https://github.com/hashicorp/nomad/pull/16743 we can finally start doing that sort of thing. But almost none of the API does yet.

Reply:
I should have noted this somewhere (either in code or the PR comment), but this line is to protect against _old_ releases, so there's no chance to break in the future (unless we _really_ want to 😬) since this message is baked in the binaries.

To better explain why this check is needed, imagine you use the 1.6.x UI code to access a cluster running Nomad 1.5.x. The 1.6.x UI code will make a request for `GET /v1/node/pools`, which the 1.5.x server interprets as a read for the node with ID `pools`, resulting in this error message.

We can change the logic for the next 1.5.x release (1.5.9 at the time of this writing), but we will still need to check for this specific error message to support any versions before that. I also used the exact error message to prevent false positives.

The other important part is that, in practice, this won't happen in a real cluster until we allow running a stand-alone UI. In a real federated cluster, accessing a region running Nomad 1.6.x from the UI of a Nomad 1.5.x region will result in the error handled in https://github.com/hashicorp/nomad/pull/18020 because the request is forwarded via RPCs.